app/src/components/LayoutHeader.vue

```diff
@@ -42,30 +42,138 @@
+        <div class="hidden md:block group-hover:text-grey-500">Cart</div>
+        <CartIcon class="icon-small icon-primary" />
+        <div class="group-hover:text-grey-500">12</div>
+      </div>
+
+      <div class="border-r border-grey-100 h-14"></div>
+
+      <div class="group relative">
+        <div class="flex items-center h-16 gap-x-2 space-x-4 group cursor-pointer">
+          <div class="hidden md:block group-hover:text-grey-500">
+            <span>{{ userDisplayName }}</span>
+          </div>
+          <figure>
+            <Jazzicon :address="userAddress" :key="userAddress" width="40" />
+          </figure>
+          <div>
+            <ArrowBottomIcon class="icon-small icon-primary" />
+          </div>
+        </div>
+        <!-- menu-->
+        <div
+          class="
+            absolute
+            hidden
+            group-hover:flex
+            z-10
+            border border-grey-400
+            p-5
+            right-2
+            md:right-20
+            bg-white
+            text-grey-400
+            flex-col
+            gap-y-2
+            uppercase
+            font-medium
+            text-left
+            whitespace-nowrap
+          "
+        >
+          <div>{{ userDisplayName }}</div>
+          <div>{{ balance }} ETH</div>
+          <div class="border-b border-grey-400 my-4"></div>
+
+          <button
+            @click="changeWallet"
+            class="cursor-pointer hover:text-grey-500 flex no-underline uppercase font-medium"
+          >
+            change wallet
+          </button>
+          <button
+            @click="disconnect"
+            class="cursor-pointer hover:text-grey-500 flex no-underline uppercase font-medium"
+          >
+            disconnect wallet
+          </button>
+        </div>
       </div>
-    </nav>
+    </div>
+    <div v-else-if="!isSupportedNetwork" class="ml-auto flex items-center">
+      <div class="flex h-12 items-center pr-2 w-12">
+        <WarningIcon class="icon stroke-pink" />
+      </div>
+      <div class="text-gray-500">Unsupported network</div>
+    </div>
+    <div v-else @click="connectWallet" class="flex items-center h-14 gap-x-2 group cursor-pointer ml-auto">
+      <div class="hidden md:block group-hover:text-grey-500">Connect</div>
+      <div>
+        <ConnectWalletIcon class="icon icon-primary" />
+      </div>
+    </div>
   </header>
 </template>
 
 <script lang="ts">
 import { defineComponent } from 'vue';
 import useWalletStore from 'src/store/wallet';
 import { WarningIcon } from '@fusion-icons/vue/interface';
+import { Wallet3Icon as ConnectWalletIcon } from '@fusion-icons/vue/web3';
+import { ArrowBottom2Icon as ArrowBottomIcon } from '@fusion-icons/vue/interface';
+import Jazzicon from 'src/components/Jazzicon.vue';
+import { Cart2Icon as CartIcon } from '@fusion-icons/vue/interface';
+import { formatEther, JsonRpcProvider, Web3Provider } from 'src/utils/ethers';
```

Comment:
A nit, but let's organize all the imports for readability. We don't have hard rule here, but for some inspiration here's how we do it in `Cart.vue`
![image](https://user-images.githubusercontent.com/17163988/131572150-edd535b0-735c-4363-b0d7-64f45ad98aae.png)

Reply:
Sure! I see that you'all use more "readability comments" than I'm used to. I'll adapt myself :)

Reply:
Heh yes IMO more comments is the way to go, especially for helping future contributors. Thanks! 🙂